src/main/kotlin/biokotlin/featureTree/TypeSchema.kt

```diff
@@ -0,0 +1,390 @@
+package biokotlin.featureTree
+
+import java.util.*
+
+// This file is incomplete and will require a near total rework due to some complexities in the underlying SO file.
+
+const val SO_PATH = "src/main/resources/featureTree/so.obo"
+private interface Type {
+    val names: Set<String>
+    val isA: Set<Type>
+    val partOf: Set<Type>
+}
+
+private class BaseSchema private constructor() {
+    val baseRegistry = mutableMapOf<String, MutableSet<BaseType>>()
+    inner class BaseType(
+        override val names: Set<String>,
+        override val isA: MutableSet<BaseType>,
+        override val partOf: MutableSet<BaseType>
+    ): Type {
+        /**
+         * INVARIANTS:
+         * 1. All names point to this in [baseRegistry]
+         */
+        fun invariants(): Boolean {
+//            names.forEach {
+//                val registration = baseRegistry[it]
+//                check(registration != null && registration.contains(this)) { "1" }
+//            }
+            return true
+        }
+
+        init {
+//            names.forEach { enroll(it, this) }
+//            assert { invariants() }
+        }
+
+        fun addPartOf(partOf: BaseType) {
+//            this.partOf.add(partOf)
+//            assert { invariants() }
+        }
+
+        fun addIsA(isA: BaseType) {
+//            this.isA.add(isA)
+//            assert { invariants() }
+        }
+    }
+    companion object {
+        private val instance = BaseSchema()
+
+        private data class ParsedData(
+            val synonyms: List<String>,
+            val isA: List<String>,
+            val partOf: List<String>
+        )
+
+        private fun enroll(name: String, type: BaseType) {
+//            val existing = instance.baseRegistry[name]
+//            if (existing == null) {
+//                instance.baseRegistry[name] = mutableSetOf(type)
+//            } else {
+//                existing.add(type)
+//            }
+        }
+        init {
+            // Populates registry with information from the so.obo file
```

Comment:
Thought I had already commented on this, but don't see the comment.  QUestion on the so.obo file.  Is this an official ontology file that pertains to all/most gff files?  Does it make sense to go through all 25K lines, or are there specific relationships we want to prioritize on the first run of this?  

This particular file is mostly commented out, so I'm wondering how the so.obo file fits in with the current code flow.  Can you explain at bit?